internal/resources/kubernetessecret/resource_secret.go

```diff
@@ -6,51 +6,78 @@ SPDX-License-Identifier: MPL-2.0
 package kubernetessecret
 
 import (
+	"context"
 	"encoding/base64"
 	"encoding/json"
 	"fmt"
 
 	"github.com/go-openapi/strfmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/customdiff"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/pkg/errors"
 
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/authctx"
+	clienterrors "github.com/vmware/terraform-provider-tanzu-mission-control/internal/client/errors"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/helper"
 	secretmodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/kubernetessecret/cluster"
+	secretexportmodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/kubernetessecret/cluster/secretexport"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/common"
 	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/kubernetessecret/scope"
 )
 
+type (
+	contextMethodKey struct{}
+)
+
 func ResourceSecret() *schema.Resource {
 	return &schema.Resource{
-		Schema: secretSchema,
+		CreateContext: resourceSecretCreate,
+		DeleteContext: resourceSecretDelete,
+		UpdateContext: resourceSecretInPlaceUpdate,
+		ReadContext:   dataSourceSecretRead,
+		Schema:        secretSchema,
+		CustomizeDiff: customdiff.All(
+			validateSpec,
+		),
 	}
 }
 
 var secretSchema = map[string]*schema.Schema{
 	NameKey: {
-		Type:     schema.TypeString,
-		Required: true,
-		ForceNew: true,
+		Type:        schema.TypeString,
+		Description: "Name of the secret resource.",
+		Required:    true,
+		ForceNew:    true,
```

Comment:
nit: Please add any validations/constraints that have been specified in the swagger file. Ex - length of the name, the format etc.

Reply:
added PTAL